### ISSUE TITLE
DateTimeImmutable support

### ISSUE THREAD

**Issue by dereuromark** [jasvrcek/ICS#68]
> Jsvrcek\ICS\Model\CalendarEvent::setStart(): Argument #1 ($start) must be of type DateTime, DateTimeImmutable given

It should also in general accept the interface IMO (also DateTimeInterface instead of concrete DateTime)
so also other libraries work:
- Carbon
- Chronos

**Comment by jasvrcek** [comment]
@dereuromark you are right, it would be better for the type hint to be DateTimeInterface. Would you mind making a pull request?
